src/pydap/net.py

```diff
@@ -29,44 +31,39 @@ def raise_for_status(response):
         )
 
 
-def follow_redirect(url, application=None, session=None,
-                    cookies_dict=None):
+def follow_redirect(url, application=None, session=None):
     """
     This function essentially performs the following command:
     >>> Request.blank(url).get_response(application)
 
     It however makes sure that the request possesses the same cookies and
     headers as the passed session.
-
-    It recursively follows 302 redirects.
```

Comment:
What, may I ask, is the key change here? You've removed the 302 following, in exchange for a HEAD request and copying the headers? If that does the job, great, but what headers does NASA/URS set that makes this possible? Would you be able to document that in the code, because it's essentially an external precondition that we'll require to be true. If that ever ceases to be true (and you're not involved in the development), that will be important for other people to know to be able to identify what has gone wrong.